cmd/kube-apiserver/app/server.go

```diff
@@ -52,6 +52,8 @@ const (
 	// Maximum duration before timing out read/write requests
 	// Set to a value larger than the timeouts in each watch server.
 	ReadWriteTimeout = time.Minute * 60
+	//TODO: This can be tightened up. It still matches objects named watch or proxy.
+	defaultLongRunningRequestRE = "(/|^)((watch|proxy)(/.*|)|(logs|portforward|exec)(/|))$"
```

Comment:
* Don't make the watch/proxy rule match to the end of the string, just ensure it is followed by `(/|$)`
* Simplify `(/|)` to `/?`

Something like:
```
"(/|^)((watch|proxy)(/|$)|(logs|portforward|exec)/?$)"
```